src/main/java/duke/Command.java

```diff
@@ -0,0 +1,183 @@
+    private Ui ui;
+    private FileLoaderSaver storage;
+    private String fullCommand;
+
+    private CommandTypes commandType;
+
+    // enum of all command types
+    public enum CommandTypes {
+        LIST,
+        TODO,
+        DEADLINE,
+        EVENT,
+        MARK,
+        UNMARK,
+        DELETE,
+        EXIT
+    }
+
+    /**
+     * Constructor for command objects
+     * @param commandType type of command this command with execute
+     * @param fullCommand full user input command in String
+     */
+    public Command (CommandTypes commandType, String fullCommand) {
+        this.commandType = commandType;
+        this.fullCommand = fullCommand;
+    }
+
+    /**
+     * Checks if command entered is EXITWORD
+     * @return true if command is EXITWORD else false
+     */
+    public boolean isExit() {
+        return fullCommand.equals(EXITWORD);
+    }
+
+    /**
+     * Executes tasks based on the command type of the command
+     * @param toDoList list containing all the current tasks
+     * @param ui user interaction object for output
+     * @param storage duke.FileLoaderSaver object for saving tasks after creating, deletion or manipulation
+     * @throws IOException
+     * @throws Exception
+     */
+    public void execute(ToDoList toDoList, Ui ui, FileLoaderSaver storage) throws IOException, Exception {
+        this.toDoList = toDoList;
+        this.ui = ui;
+        this.storage = storage;
+
+        switch (commandType) {
+            case LIST:
+                toDoList.listTasks();
+                break;
+            case TODO:
+                addToDo(fullCommand);
+                storage.writeToFile(toDoList.createTxtFile());
+                toDoList.displayListSize();
+                break;
+            case DEADLINE:
+                addDeadline(fullCommand);
+                storage.writeToFile(toDoList.createTxtFile());
+                toDoList.displayListSize();
+                break;
+            case EVENT:
+                addEvent(fullCommand);
+                storage.writeToFile(toDoList.createTxtFile());
+                toDoList.displayListSize();
+                break;
+            case MARK:
+            case UNMARK:
+                changeMark(fullCommand);
+                storage.writeToFile(toDoList.createTxtFile());
+                break;
+            case DELETE:
+                deleteTask(fullCommand);
+                storage.writeToFile(toDoList.createTxtFile());
+                toDoList.displayListSize();
+                break;
+            case EXIT:
+                Ui.goodBye();
+                break;
+        }
+    }
+
+    /**
+     * Changes status of the task according to index given
+     *
+     * @param command change mark command from user
+     */
+    private void changeMark(String command) {
+        String[] splitComm = command.split(" ");
+        String action = splitComm[0];
+        int index = Integer.parseInt(splitComm[1]) - 1;
```

Comment:
What if `parseInt` throw an exception? 

Even though the number is checked with regex during parsing, I feel it might be better to handle the exception.
Or even better approach is to make the Parser also parse the index, and call this method with the index. This way all the parsing will be done in one place and the exceptions related to parsing can also be thrown from one place.